src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -0,0 +1,985 @@
+
+    // This disables storage of SSID details in NVS which has been shown to be
+    // problematic at times for the ESP32, it is safer to always pass fresh
+    // config and have the ESP32 resolve the details at runtime rather than
+    // use a cached set from NVS.
+    esp_wifi_set_storage(WIFI_STORAGE_RAM);
+
+    // Configure the SSID details for the station based on the SSID and
+    // password provided to the Esp32WiFiManager constructor.
+    wifi_config_t conf;
+    memset(&conf, 0, sizeof(wifi_config_t));
+    strcpy(reinterpret_cast<char*>(conf.sta.ssid), ssid_);
+    // if we have a password check if it is a PSK or regular password
+    if(password_)
+    {
+        if (strlen(password_) == 64) // it's the PSK and not a password.
+        {
+            memcpy(reinterpret_cast<char*>(conf.sta.password), password_, 64);
+        }
+        else
+        {
+            strcpy(reinterpret_cast<char*>(conf.sta.password), password_);
+        }
+    }
+
+    LOG(INFO, "[WiFi] Configuring WiFi stack");
+    ESP_ERROR_CHECK(esp_wifi_set_config(ESP_IF_WIFI_STA, &conf));
+
+    // Attempt to connect to the SSID, this will block until the ESP32 starts
+    // the connection process, note it may not have an IP address immediately
+    // thus the need to check the connection result a few times before giving
+    // up with a FATAL error.
+    LOG(INFO, "[WiFi] Starting WiFi stack");
+    ESP_ERROR_CHECK(esp_wifi_start());
+
+    uint8_t attempt = 0;
+    EventBits_t bits;
+    uint32_t bitMask = WIFI_CONNECTED_BIT;
+    while(++attempt <= MAX_CONNECTION_CHECK_ATTEMPTS)
+    {
+        // If we have connected to the SSID we then are waiting for DHCP.
+        if(bits & WIFI_CONNECTED_BIT)
+        {
+            LOG(INFO, "[DHCP] [%d/%d] Waiting for IP address assignment.",
+                attempt, MAX_CONNECTION_CHECK_ATTEMPTS);
+        }
+        else
+        {
+            // Waiting for SSID connection
+            LOG(INFO, "[WiFi] [%d/%d] Waiting for SSID connection.",
+                attempt, MAX_CONNECTION_CHECK_ATTEMPTS);
+        }
+        bits = xEventGroupWaitBits(wifiStatusEventGroup_,
+            bitMask,        // bits we are interested in
+            pdFALSE,        // clear on exit
+            pdTRUE,         // wait for all bits
+            WIFI_CONNECT_CHECK_INTERVAL);
+        // Check if have connected to the SSID
+        if(bits & WIFI_CONNECTED_BIT)
+        {
+            // Since we have connected to the SSID we now need to track that we
+            // get an IP via DHCP.
+            bitMask |= DHCP_GOTIP_BIT;
+        }
+        // Check if we have received an IP via DHCP.
+        if(bits & DHCP_GOTIP_BIT)
+        {
+            break;
+        }
+    }
+
+    // Check if we successfully connected or not. If not, force a reboot.
+    if((bits & WIFI_CONNECTED_BIT) != WIFI_CONNECTED_BIT)
+    {
+        LOG(FATAL, "[WiFi] Failed to connect to SSID.");
```

Comment:
please also print the actual ssid we were trying to connect to

Reply:
fixed